ext/intl/grapheme/grapheme_string.c

```diff
@@ -816,4 +816,85 @@ PHP_FUNCTION(grapheme_extract)
 	RETURN_STRINGL(((char *)pstr), ret_pos);
 }
 
+PHP_FUNCTION(grapheme_str_split)
+{
+	char *pstr, *end;
+	zend_string *str;
+	zend_long split_len = 1;
+
+	unsigned char u_break_iterator_buffer[U_BRK_SAFECLONE_BUFFERSIZE];
+	UErrorCode ustatus = U_ZERO_ERROR;
+	int32_t pos, current, i, end_len = 0;
+	UBreakIterator* bi;
+	UText *ut = NULL;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_STR(str)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(split_len)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (split_len <= 0) {
+		zend_argument_value_error(2, "must be greater than 0");
+		RETURN_THROWS();
+	} else if (split_len > UINT_MAX / 4) {
```

Comment:
Is there a reason why this is limited to `UINT_MAX / 4`? I get that it's likely to be on the safer side for the 4 bytes max a Unicode code-point might take. I tested with this removed, but it seemed to work well.

Reply:
That's right. But I think two reasons:

- Followed `mb_str_split`
- https://3v4l.org/XbDHU seems can't exit successfully

Therefore, I think follow to `UINT_MAX / 4` by `grapheme_str_split`.

Reply:
Lines from 837 to 840 are wrong, you likely forgot to remove them after the feedback.

Reply:
@devnexen Excuse me, Could you tell me what is wrong?

Reply:
Look at it this way

```c
    if (split_len <= 0) {
		zend_argument_value_error(2, "must be greater than 0");
		RETURN_THROWS();
	} else if (split_len > UINT_MAX / 4) {
    if (split_len <= 0 || split_len > UINT_MAX / 4) {
		zend_argument_value_error(2, "must be greater than 0 and less than or equal to %d", UINT_MAX / 4);
		RETURN_THROWS();
	}
		RETURN_THROWS();
	}
```

Reply:
should be just

```c
if (split_len <= 0 || split_len > UINT_MAX / 4) {
		zend_argument_value_error(2, "must be greater than 0 and less than or equal to %d", UINT_MAX / 4);
		RETURN_THROWS();
	}
```

what do you think ?

Reply:
Oh! I overlooked it! Thank you! fixed.